### PR TITLE
[16.0][MIG] base_multi_company

### PR DESCRIPTION
**Pull request by atchuthan**
cc: @SodexisTeam

### CODE REVIEW

**Comment by amkarthik** [comment]
@gurneyalex 
Test is being failed for the following reason
ModuleNotFoundError: No module named 'odoo_test_helper'.
Can you please install the needed module

Thanks

**Comment by yajo** [comment]
The problem comes from https://github.com/OCA/multi-company/pull/422. Please add that dependency to `test-requirements.txt` just like it's done in that PR, and tests will work again.

**Comment by rafaelbn** [comment]
@atchuthan @amkarthik all test green!

**Review by rafaelbn — COMMENTED** [review]
@rousseldenis @Cedric-Pigeon would you like to review 😄 as you know this module? 👍

**Review by amkarthik — APPROVED** [review]

**Comment by rafaelbn** [comment]
/ocabot migration base_multi_company

ping @etobella

**Review by yajo — CHANGES_REQUESTED** [review]
I'm missing these PRs in this migration: https://github.com/OCA/multi-company/pull/354, https://github.com/OCA/multi-company/pull/490

I'm fwporting the latter one to 15.0 in https://github.com/OCA/multi-company/pull/503, but those were present in 14.0 and are important fixes.

**Comment by rafaelbn** [comment]
Hello @atchuthan , please could you review comments from @yajo ? Thank you! 😄

**Comment by odoo-sh** [comment]
@yajo added the PR's #354 #490

**Comment by amkarthik** [comment]
[@yajo](https://github.com/yajo) added the PR's [#354](https://github.com/OCA/multi-company/pull/354) [#490](https://github.com/OCA/multi-company/pull/490)

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
I think you should rewrite the commit message for dedc8d8, as it seems it should be merged.

**Comment by amkarthik** [comment]
@yajo As it is test requirement, Can we remove it?

**Review by yajo — CHANGES_REQUESTED** [review]
No, you don't have to remove the file. You just have to rename the commit, because having `DO NOT MERGE` in a commit that must be merged is misleading.

Please also rename this commit https://github.com/OCA/multi-company/pull/458/commits/c1e6e2055c1728e30510a31f0ec42f62b0a05cdc because it is also misleading. It makes the reader think that a commit with the name `[MIG] base_multi_company: black, isort, prettier` just contains cosmetic changes, but you're actually doing the manual changes there. Instead, explain those changes appropriately in the commit message.

**Comment by amkarthik** [comment]
@yajo Modified the commit message.

**Review by amkarthik — APPROVED** [review]

File: base_multi_company/hooks.py
Comment (SirAionTech) [inline base_multi_company/hooks.py]: This should be removed because it is deprecated and raises a WARNING in the logs when executed, see https://github.com/odoo/odoo/blame/7fbde13b0c6bd9b265f267f77ce3ccc6874b42e9/odoo/api.py#L493.

**Comment by hgarridoco** [comment]
@amkarthik @yajo any updates for this PR?

**Comment by rafaelbn** [comment]
You need a @OCA/intercompany-maintainers I guess 😄

**Comment by pedrobaeza** [comment]
Please include #536

File: base_multi_company/hooks.py
Comment (pedrobaeza) [inline base_multi_company/hooks.py]: You can improve it this way:
```suggestion
                "[('company_ids', 'in', [False] + company_ids)]"
```

File: base_multi_company/hooks.py
Comment (pedrobaeza) [inline base_multi_company/hooks.py]: Improvement:
```suggestion
                " [('company_id', 'in', [False, user.company_id.id])]"
```

File: base_multi_company/models/multi_company_abstract.py
Comment (pedrobaeza) [inline base_multi_company/models/multi_company_abstract.py]: Why removing this?

Comment (odoo-sh) [reply base_multi_company/models/multi_company_abstract.py]: @pedrobaeza 
we removed it according to the PR #354

File: base_multi_company/hooks.py
Comment (SodexisTeam) [reply base_multi_company/hooks.py]: @pedrobaeza 
work done on the improvement

File: base_multi_company/hooks.py
Comment (SodexisTeam) [reply base_multi_company/hooks.py]: @pedrobaeza 
work done on the improvement

File: base_multi_company/models/multi_company_abstract.py
Comment (SodexisTeam) [reply base_multi_company/models/multi_company_abstract.py]: @pedrobaeza 
we removed the default according to the PR #354

**Review by SirAionTech — CHANGES_REQUESTED** [review]
Thanks for the PR!
I have tried this one with https://github.com/OCA/multi-company/pull/460 and everything seems to work correctly.
About the code, I just happened to notice what below, I haven't looked at all the changes.

File: base_multi_company/hooks.py
Comment (SodexisTeam) [reply base_multi_company/hooks.py]: @SirAionTech 
removed

**Review by SirAionTech — APPROVED** [review]
Thanks for the PR!
I have tried this one with https://github.com/OCA/multi-company/pull/460 and everything seems to work correctly.
Other than https://github.com/OCA/multi-company/pull/458#discussion_r1409549046 I haven't looked at the code changes though.

**Comment by pedrobaeza** [comment]
Please include this hotfix as well: #540

You can squash both in the same commit for reducing the commit history, as it's not relevant per se.

**Review by pedrobaeza — APPROVED** [review]
Thanks for the work

/ocabot merge nobump

**Pull request merged**
